test/templates.post.test.js

```diff
@@ -496,6 +496,223 @@ describe('POST templates', () => {
+              }
+            ]
+          }
+        ]
+      }
+    });
+
+    findTemplateByName.mockReturnValue(null);
+    fetchUrl.mockReturnValue('');
+    const templateName = '@adobe/developer-console-template';
+    const template = {
+      id: '56bf8211-d92d-44ef-b98b-6ee89812e1d7',
+      name: templateName,
+      description: 'Developer Console template',
+      status: 'InVerification',
+      version: '1.0.0',
+      links: {
+        consoleProject: DEVELOPER_CONSOLE_PROJECT
+      },
+      author: 'Capernicus'
+    };
+    addTemplate.mockReturnValue(template);
+    const response = await action.main({
+      IMS_URL: process.env.IMS_URL,
+      IMS_CLIENT_ID,
+      IMS_CLIENT_SECRET,
+      IMS_AUTH_CODE,
+      IMS_SCOPES,
+      TEMPLATE_REGISTRY_ORG: process.env.TEMPLATE_REGISTRY_ORG,
+      TEMPLATE_REGISTRY_REPOSITORY: process.env.TEMPLATE_REGISTRY_REPOSITORY,
+      ACCESS_TOKEN_GITHUB: process.env.ACCESS_TOKEN_GITHUB,
+      TEMPLATE_REGISTRY_API_URL: process.env.TEMPLATE_REGISTRY_API_URL,
+      __ow_method: HTTP_METHOD,
+      name: DEVELOPER_CONSOLE_TEMPLATE_NAME,
+      links: {
+        consoleProject: DEVELOPER_CONSOLE_PROJECT
+      },
+      description: 'Developer Console template',
+      version: '1.0.0',
+      createdBy: 'Capernicus',
+      author: 'Capernicus',
+      ...fakeParams
+    });
+    expect(response).toEqual({
+      statusCode: 200,
+      body: {
+        ...template,
+        _links: {
+          self: {
+            href: `${process.env.TEMPLATE_REGISTRY_API_URL}/templates/${templateName}`
+          }
+        }
+      }
+    });
+    expect(mockLoggerInstance.info).toHaveBeenCalledWith('Calling "POST templates"');
+    expect(validateAccessToken).toHaveBeenCalledWith(IMS_ACCESS_TOKEN, process.env.IMS_URL, IMS_CLIENT_ID);
+    expect(generateAccessToken).toHaveBeenCalledWith(IMS_AUTH_CODE, IMS_CLIENT_ID, IMS_CLIENT_SECRET, IMS_SCOPES);
+    expect(findTemplateByName).toHaveBeenCalledWith({}, DEVELOPER_CONSOLE_TEMPLATE_NAME);
+    expect(addTemplate).toHaveBeenCalledWith({ MONGODB_NAME: undefined, MONGODB_URI: undefined }, {
+      name: DEVELOPER_CONSOLE_TEMPLATE_NAME,
+      apis: [
+        {
+          credentialType: 'serviceAccount',
+          flowType: 'oauth2',
+          code: 'AdobeIO',
+          productProfiles: undefined
+        }
+      ],
+      credentials: [
+        {
+          type: 'serviceAccount',
+          flowType: 'oauth2'
+        }
+      ],
+      links: {
+        consoleProject: DEVELOPER_CONSOLE_PROJECT
+      },
+      description: 'Developer Console template',
+      version: '1.0.0',
+      createdBy: 'Capernicus',
+      author: 'Capernicus'
+    });
+    // TODO: Uncomment the following after integrating with App Builder templates again
+    // expect(createReviewIssue).toHaveBeenCalledWith(TEMPLATE_NAME, TEMPLATE_GITHUB_REPO, process.env.ACCESS_TOKEN_GITHUB, process.env.TEMPLATE_REGISTRY_ORG, process.env.TEMPLATE_REGISTRY_REPOSITORY);
```

Comment:
perhaps we should add a story for this so we don't forget after

Reply:
Good thinking. Just added one to the backlog